pywbem_mock/_instancewriteprovider.py

```diff
@@ -270,367 +209,280 @@ def __init__(self, cimrepository=None):
-
-        The input parameters must have been validated before this method
-        is called including:
-
-        1. Namespace is valid in CIM repository.
-        2. The ``NewInstance`` is a valid CIM Instance.
-        3. All key properties defined in the class  of ``NewInstance`` are
-           in ``NewInstance``. This method cannot add properties or
-           property values to ``NewInstance``.
-        4. There are no properties in ``NewInstance`` that are not in the
-           corresponding class.
-        5. NewInstance path must not exist in the CIM repository.
+        Create a new CIM instance in the CIM repository of the mock WBEM server.
+
+        Validation already performed by the provider dispatcher that calls
+        this provider method:
+        - The provider method is called only for the registered class and
+          namespace.
+        - The Python types of all input parameters to this provider method are
+          as specified below.
+        - The namespace exists in the CIM repository.
+        - The creation class of the new instance exists in the namespace
+          in the CIM repository.
+        - All properties specified in the new instance are exposed (i.e.
+          defined and inherited with any overrides resolved) by the creation
+          class in the CIM repository, and have the same type-related
+          attributes (i.e. type, is_array, embedded_object).
+
+        Validation that should be performed by this provider method:
+        - new_instance does not specify any properties that are not
+          allowed to be initialized by the client, depending on the model
+          implemented by the provider.
+        - new_instance specifies all key properties needed by the
+          provider, depending on the model implemented by the provider.
+          The CIM repository will reject any new instance that does not have
+          all key properties specified.
+        - The new instance does not exist in the CIM repository.
+          This validation needs to be done by the provider because the
+          determination of the key properties for the instance path may depend
+          on the model implemented by the provider.
+          The CIM repository will reject the creation of instances that already
+          exist, so this check can be delegated to the repository once the
+          instance path has been determined.
 
         Parameters:
 
           namespace (:term:`string`):
-            The name of the CIM namespace in the CIM repository (case
-            insensitive). Must not be `None`. Leading or trailing slash
-            characters are ignored.
+            The name of the CIM namespace in which the CIM instance is to be
+            created, in any lexical case, and with leading and trailing slash
+            characters removed.
 
-          NewInstance (:class:`~pywbem.CIMInstance`):
+          new_instance (:class:`~pywbem.CIMInstance`):
```

Comment:
Why did we switch from NewInstance to new_instance for the provider?

Reply:
Yes, I did that as an indication that this is no longer the original NewInstance parameter provided to the client operation (e.g. lexical case of property names has been adjusted).